packages/flutter/lib/src/cupertino/text_selection_toolbar.dart

```diff
@@ -366,26 +360,30 @@ class _RenderCupertinoTextSelectionToolbarShape extends RenderShiftedBox {
       return path..addRRect(rrect);
     }
 
-    final Offset localAnchor = globalToLocal(_anchor);
-    final double arrowTipX = clampDouble(
-      localAnchor.dx,
-      _kToolbarBorderRadius.x + _kToolbarArrowSize.width / 2,
-      size.width - _kToolbarArrowSize.width / 2 - _kToolbarBorderRadius.x,
-    );
+    final BoxParentData childParentData = child.parentData! as BoxParentData;
+    final Offset localAnchor = globalToLocal(isAbove ? _anchorAbove : _anchorBelow);
+
+    final double centerX = childParentData.offset.dx + child.size.width / 2;
+    final double arrowXOffsetFromCenter = localAnchor.dx - centerX;
+    final double arrowTipX = child.size.width / 2 + arrowXOffsetFromCenter;
 
     // Draw the path clockwise, starting from the beginning side of the arrow.
-    if (_isAbove) {
+    if (isAbove) {
+      final double arrowBaseY = child.size.height - _kToolbarArrowSize.height;
+      final double arrowTipY = child.size.height;
```

Comment:
Looks like the path is now described in `child`'s coordinates? If so could you update the comment before this method?